tests/system/test_aws_s3.py

```diff
@@ -0,0 +1,104 @@
+# Copyright 2023 Iguazio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+import random
+
+import pandas as pd
+import pytest
+
+import mlrun
+import mlrun.errors
+import mlrun.feature_store as fstore
+from mlrun.datastore.datastore_profile import (
+    DatastoreProfileS3,
+    register_temporary_client_datastore_profile,
+)
+from mlrun.datastore.targets import ParquetTarget
+from mlrun.utils import logger
+from tests.system.base import TestMLRunSystem
+
+# Used to test dataframe functionality (will be saved as csv)
+test_df_string = "col1,col2,col3\n1,2,3"
+
+test_environment = TestMLRunSystem._get_env_from_file()
+aws_keys = ["AWS_ACCESS_KEY_ID", "AWS_SECRET_ACCESS_KEY", "AWS_BUCKET_NAME"]
+
+
+@TestMLRunSystem.skip_test_if_env_not_configured
+@pytest.mark.skipif(
+    not all(test_environment[key] for key in aws_keys),
+    reason=f"one of {aws_keys} keys is not set, skipping until testing against real aws",
+)
```

Comment:
This skipif doesn't work as it should. I get
```
KeyError: 'AWS_ACCESS_KEY_ID'
```
when I try to run this suite without setting the env vars.

Also, doing it like this will result in the user not knowing which variable is missing. This matters if we get this failure in CI and we can't just check it in the debugger.

I suggest:
```suggestion

@TestMLRunSystem.skip_test_if_env_not_configured
@pytest.mark.skipif(
    not test_environment.get("AWS_ACCESS_KEY_ID"),
    reason=f"AWS_ACCESS_KEY_ID is not set",
)
@pytest.mark.skipif(
    not test_environment.get("AWS_SECRET_ACCESS_KEY"),
    reason=f"AWS_SECRET_ACCESS_KEY is not set",
)
@pytest.mark.skipif(
    not test_environment.get("AWS_BUCKET_NAME"),
    reason=f"AWS_BUCKET_NAME is not set",
)
```

Reply:
Will do